### PR TITLE
fix: Polygon simplification on E&A page

### PR DESCRIPTION
**Pull request by hanbyul-here**
Close https://github.com/NASA-IMPACT/veda-ui/issues/1267, Close https://github.com/US-GHG-Center/veda-config-ghg/issues/648

### Description of Changes

Increased the limit of the number of each polygon.
Separated out a function that handles the aoi validation for an easy test and better readability.
Fix error handling.

### Notes & Questions About Changes

The number that I ended up with is an outcome of my experiment with our GHG endpoint, details are in : https://github.com/US-GHG-Center/veda-config-ghg/issues/648#issuecomment-2487390157

I think our current setup was more for multiple simple polygons - and this setup is more for 1 complicated polygon (which also hopefully covers the multiple simple polygons case.)

### Validation / Testing
We should test with GHG endpoint because of AWS setup there, I made a preview link : https://deploy-preview-664--ghg-demo.netlify.app/exploration

### CODE REVIEW

**Comment by aboydnw** [comment]
hm, I'm not able to get anything to load. Here are the files I tried:

[Alabama_State_Boundary-shp.zip](https://github.com/user-attachments/files/17849838/Alabama_State_Boundary-shp.zip)
[bs.shp.zip](https://github.com/user-attachments/files/17849849/bs.shp.zip)
[alabama state geojson](https://drive.google.com/file/d/1a2dEP3_UQWil2wkSCVfFNbqgCbaMQeop/view?usp=drive_link)
[Indianapolis geojson](https://drive.google.com/file/d/1GEq8BjWoJV185bb0M_cGhbVYXasse32T/view?usp=drive_link)

Each time I get this error: 
image

**Comment by hanbyul-here** [comment]
@aboydnw Thanks for testing and catching the bug that I pushed at the last imnute 😓 Please test it again when you have a chance.

**Comment by aboydnw** [comment]
This seems to be working much better now! The shape for Indianapolis still doesn't show exactly accurate, but now it seems to be because rings are not supported, which I think is an acceptable limitation.

image

image


I was able to upload a fairly complex shapefile and it worked well. I think this should give folks at AGU more ability to find a shapefile that works @Jeanne-le-Roux @slesaad 
image

**Review by aboydnw — APPROVED** [review]
From a manual testing perspective, looks good to me.

File: app/scripts/components/common/map/controls/hooks/use-custom-aoi.tsx [hunk 1]
```diff
@@ -7,42 +7,46 @@ import { multiPolygonToPolygons } from '../../utils';
...
-const eachFeatureMaxPointNum = 100;
+const maxTolerance = 5;
+
+export const eachFeatureMaxPointNum = 500;
+export const maxPolygonNum = 200;
 export const acceptExtensions = extensions.map((ext) => `.${ext}`).join(', ');
 
+export const INVALID_GEOMETRY_ERROR =
+  'Wrong geometry type. Only polygons or multi polygons are accepted.';
+export const TOO_MANY_POLYGON_ERROR =
```
[the remark this answers is not in the document]
Comment (hanbyul-here) [reply app/scripts/components/common/map/controls/hooks/use-custom-aoi.tsx]: I made my own commit since there are other places using this const! thanks !

**Pull request merged**